frontend/public/components/routes/utils.ts

```diff
@@ -0,0 +1,144 @@
+import * as _ from 'lodash-es';
+import * as yup from 'yup';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { EditorType } from '@console/shared/src/components/synced-editor/editor-toggle';
+import { getActiveNamespace } from '../../actions/ui';
+import { AlternateServiceEntryType, RouteFormProps, UNNAMED_PORT_KEY } from './create-route';
+
+export const routeValidationSchema = yup.object({
+  editorType: yup.string(),
+  formData: yup.object().when('editorType', {
+    is: EditorType.Form,
+    then: yup.object({
+      name: yup.string().required(),
+      service: yup.string().required(),
+      targetPort: yup.string().required(),
+      termination: yup.string().when('formData.secure', {
+        is: true,
+        then: yup.string().required(),
+      }),
+    }),
+  }),
+});
+
+export const convertRouteToEditForm = (
+  services: K8sResourceKind[],
+  data: K8sResourceKind,
+): Partial<RouteFormProps> => {
+  if (!data) {
+    return null;
+  }
+  const { metadata, spec } = data;
+  return {
+    name: metadata?.name,
```

Comment:
if data is present then metadata will always be there right in case or Route, if that's true then we can return above otherwise